cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,376 @@
+
+3. Create a Radius environment and add an AWS cloud provider.
+    ```
+    rad group create default
+
+    rad env create default
+    
+    rad env update default --aws-account-id <aws-account-id> --aws-region <aws-region> 
+    ```
+
+4. Register the role ARN using `rad credential register aws irsa`.
+    ```
+   rad credential register aws irsa --iam-role <roleARN>
+    ```
+
+Radius is now configured to use AWS IRSA for authentication.
+
+## Design
+
+### High Level Design for IRSA
+
+During installation, Radius should allow user to enable IRSA for AWS while conifguring AWS provider. Then the user should be able to enter the roleARN associated with desired policies which will be stored as the AWS credential.
+
+At a high level, the full flow to setup IRSA looks as below:
+
+1. configure the cluster with oidc provider. have the role-arn configured with desired policies ready.
+2. take steps to allow  `radius-role-arn` and `eks cluster : namespace: service` to trust each other using aws portal.
+3. use [interactive](#user-story-1) approach or [non-interative](#user-story-2) to configure Radius to use IRSA
+4. during rad deploy, radius assumes the role that is configured in the credential. 
+
+OIDC is configured for a cluster and not specific to a role-arn. Therefore, once we have support for multiple crdentials in Radius, the same solution would work for supporting multitenancy.
+
+More details about each step is covered in [Detailed Design](#detailed-design)
+
+IRSA works the same way on EKS as well as non-EKS clusters.  
+
+### Architecture Diagram
+
+## User Experience/ CLI Design
+
+Following commands should be updated for AWS IRSA support
+
+`rad init` should be updated to allow configuring a AWS IRSA provider.
+
+`rad credential` command should be updated to register IAM RoleARN.
+
+`rad install kubernetes --set` should be able to enable IRSA. 
+
+
+**Sample Output:**
+
+n/a
+
+**Sample Recipe Contract:**
+
+n/a
+
+### Detailed Design
+
+#### Proposed Option
+
+Radius should support AWS IRSA.
+
+#### Prerequisite
+
+1. [Setup OIDC provider] (https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html) should be complete for the cluster. 
+   
+2. `radius-role` with desired policies should be created. Its trust relation should look similar to below. This establishes a trust relation between the annotated service-accounts in the specified namespace in specified cluster with the specified OIDC provider to be established. 
+  ```
+  {
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Effect": "Allow",
+        "Principal": {
+          "Federated": "arn:aws:iam::${ACCOUNT_ID}:oidc-provider/${OIDC_PROVIDER}"
+        },
+        "Action": "sts:AssumeRoleWithWebIdentity",
+        "Condition": {
+          "StringEquals": {
+            "${OIDC_PROVIDER}:aud": "<AUDIANCE-NAME>",
```

Comment:
For audience type, is anything else apart from `sts.amazonaws.com` supported?

```suggestion
            "${OIDC_PROVIDER}:aud": "<AUDIENCE-NAME>",
```

Reply:
need not templatize sts.